pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -286,6 +309,90 @@ func (m *ManifestWorkController) applyUnstructrued(data []byte, owner metav1.Own
 	return actual, true, err
 }
 
+func (m *ManifestWorkController) applyOwner(
+	gvr schema.GroupVersionResource,
+	namespace, name string,
+	deleteOption *workapiv1.DeleteOption,
+	owner metav1.OwnerReference,
+	recorder events.Recorder) error {
+	existing, err := m.spokeDynamicClient.
+		Resource(gvr).
+		Namespace(namespace).
+		Get(context.TODO(), name, metav1.GetOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	owners, _ := manageOwnerRef(gvr, namespace, name, deleteOption, existing.GetOwnerReferences(), owner)
+
+	if equality.Semantic.DeepEqual(owners, existing.GetOwnerReferences()) {
+		return nil
+	}
+
+	existing.SetOwnerReferences(owners)
+	_, err = m.spokeDynamicClient.
+		Resource(gvr).
+		Namespace(namespace).
+		Update(context.TODO(), existing, metav1.UpdateOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	recorder.Eventf(fmt.Sprintf(
+		"Owner of %s Updated", existing.GetKind()), "Updated %s/%s", existing.GetNamespace(), existing.GetName())
+
+	return nil
+}
+
+func manageOwnerRef(
```

Comment:
It seems like this logic could be called during our normal create flow instead of doubling our number of calls.

Reply:
We can do that with create, but not  update. Since applyDirectly does not merge OwnerRefs today. In that way the flow will be like:

1. Get resources at first to get current owners.
2. MergeOwners in normal apply
3. Get resources again to check if owners are correct.
4. Update owners if it is not correct.